reducer/count.go

```diff
@@ -7,20 +7,28 @@ import (
 )
 
 type CountProto struct {
-	target   string
-	resolver expr.FieldExprResolver
+	target              string
+	resolver, tresolver expr.FieldExprResolver
 }
 
 func (cp *CountProto) Target() string {
 	return cp.target
 }
 
+func (cp *CountProto) TargetResolver() expr.FieldExprResolver {
+	return cp.tresolver
+}
+
 func (cp *CountProto) Instantiate() Interface {
 	return &Count{Resolver: cp.resolver}
 }
 
-func NewCountProto(target string, resolver expr.FieldExprResolver) *CountProto {
-	return &CountProto{target, resolver}
+func NewCountProto(target string, tresolver, resolver expr.FieldExprResolver) *CountProto {
+	return &CountProto{
+		target:    target,
+		resolver:  resolver,
+		tresolver: tresolver,
+	}
```

Comment:
We usually put constructors immediately after the type definition, not after its methods.